src/hooks/useApi.ts

```
@@ -0,0 +1,331 @@
+
+type HttpMethod = 'GET' | 'POST' | 'PUT' | 'DELETE';
+
+type ApiConfig<
+  P extends Record<string, unknown>,
+  M extends HttpMethod = 'GET' | 'DELETE'
+> = {
+  queryParams?: Record<string, string>;
+  impersonationData?: ImpersonationData;
+  additionalHeaders?: Record<string, string>;
+  version?: string;
+} & (M extends 'POST' | 'PUT' ? { payload: P } : {});
```

Biome (1.9.4)

[error] Don't use '{}' as a type.

Prefer explicitly define the object shape. '{}' means "any non-nullable value".

(lint/complexity/noBannedTypes)